src/Views/Components/AdminDetailsPage/utils.ts

```diff
@@ -42,3 +44,38 @@ export function formatDateLocal(dateString: string) {
 
     return `${year}-${month}-${day}`;
 }
+
+/**
+ * @unreleased
+ */
+export function filterOutReadOnlyFields(
```

Comment:
@pauloiankoski nice job! couple thoughts here:

1. I think this is generic enough to RHF and REST schema that will be useful outside of just this component directory -  so we should elevate it perhaps up into `src/Admin/`
2. Since the actual filtering of `readonly` is pretty contextual and is kind of a small part here, I would probably rename this function to be like `prepareEntityRecordFromSchema()` and try to simplify the logic a bit to be easier to understand perhaps with some small utility functions inside for checking `isReadOnly()`, `isObject()`, etc.  Then, if we need to update our filtering in the future we have a dedicated place within this function to do so since it's already parsing the record.